### ISSUE TITLE
Goal 10.2: Fix Up Evaluation Script

### ISSUE THREAD

**Issue by yumna-tree** [bschink/pocket-narrator#35]
- log the generated results into wandb with metrics as columns and rows as individual logs
- take the same (use seeding) 50k (1/5 of the dataset) stories that we took during training to calculate average score evaluation per each model
- one evaluation script run results in a table of 50k lines and end scores

**Comment by yumna-tree** [comment]
my alternative kind of doesnt make any sense because that 'same test set' was never a dataset the models were trained with to begin with

**Comment by yumna-tree** [comment]
1. evaluation script
    1. takes a model which is either
        1. ngram?
        2. transformer
        3. mamba?
    2. takes a data set which is either
        1. any dataset, could be these (most likely number 1):
            1. a test set never seen before
            2. any dataset (as whole) a model was trained with
            3. the validation set of a dataset a model was trained with —> gotta figure out a way to acquire this outside of this script
    2. splits into prompt and ground truth (in the middle 
    3. feeds the splitted prompts into specified trained model
    4. gets its outputs
    5. runs all possible evaluation metrics on the prompt-output pairs
        1. distinct n
        2. repetition rate
        3. grammar
        4. llm judge
        5. text quality (coherence and cohesion)
        6. noun carryover 
        7. plain statistics word and sentence count
        8. rouge
        9. bleu
        10. perplexity
    6. logs everything into wandb in a big table with the following columns
        1. what model it was
        2. what dataset/validation set it was
        3. what the prompts were
        4. what the output were
        5. what the 6 scores were on each of those prompt output pairs
        6. final scores
    7. wandb maintains a dot plot per score category showing how each of the evaluation runs did  → also a json artifact with ALL evaluation results per story like in the dataset evaluation script
3. evaluation config file to configure:
    1. model in use
    2. data in use
    3. which metrics to measure




1. evaluate dataset script
    1. takes a dataset as input
        1. any dataset, could be these (most likely number 2 and 3):
            1. a test set never seen before
            2. any dataset (as whole) a model was trained with
            3. the validation set of a dataset a model was trained with —> gotta figure out a way to acquire this outside of this script
    2. runs all possible evaluation metrics (except rouge, bleu and perplexity) on each of the stories
        1. distinct n
        2. repetition rate
        3. grammar
        4. llm judge
        5. text quality (coherence and cohesion)
        6. noun carryover → only this eval metric would require a splitting mechanism of prompt and ground truth
        7. plain statistics word and sentence count
    3. logs everything into wandb in a big table with the following columns
        1. what dataset/validation set it was
        2. what the prompts were
        3. what the output were
        4. what the 6 scores were on each of those prompt output pairs
        5. final scores
    4. wandb maintains a dot plot per score category showing how each of the evaluation runs did → also a json artifact with ALL evaluation results per story
2. evaluate dataset config file to configure:
    1. data in use
    2. which metrics to measure



_______________________

finished the above scripts but need input on what metric to compute on what (from midpoint to end vs from beginning of story to end)
